cap/modules/repoimporter/git_importer.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of CERN Analysis Preservation Framework.
+# Copyright (C) 2018 CERN.
+#
+# CERN Analysis Preservation Framework is free software; you can redistribute
+# it and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# CERN Analysis Preservation Framework is distributed in the hope that it will
+# be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with CERN Analysis Preservation Framework; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+
+"""Github/Gitlab importer class."""
+
+from github import Github
+from gitlab import Gitlab
+from utils import parse_url, get_access_token
+
+
+class GitImporter:
+    """Github / Gitlab importer class."""
+
+    def __init__(self, host=None, owner=None,
+                 repo=None, branch='master', url=None):
+        """Initialize an importer and extract the attributes needed."""
+        if all([host, owner, repo]):
+            self.host = host.strip('/')
+            self.owner = owner
+            self.repo = repo
+            self.branch = branch
+        elif url:
+            url_attrs = parse_url(url)
+            self.host, self.owner, self.repo, self.branch = \
+                (url_attrs['host'], url_attrs['owner'],
+                 url_attrs['repo'], url_attrs['branch'])
+        else:
+            raise ValueError(
+                'Missing git attributes/url. Cannot construct instance.')
+
+        # git related attributes, can be used separately
+        repo_name = '{}/{}'.format(self.owner, self.repo)
```

Comment:
Update construction of filename to eliminate conflicts:
- repos `<repo_owner>_<repo_name>_<ref>`
- files `<repo_owner>_<repo_name>_<ref>_<filename_with_extension>`

Remove colon `:` from filename

In the future we will add file-tags for better display names, etc